### ISSUE TITLE
valid-expect-in-promise doesn't work with Promise.all

### ISSUE THREAD

**Issue by realityking** [jest-community/eslint-plugin-jest#219]
I have a test case that relies on multiple promises being tested with `Promise.all`:

```js
test('Resource loader: client usage', function () {
  expect.assertions(2);

  const quxLoader = ResourceLoader.create(['qux']);
  let promiseResolve;
  fetchResult = new Promise(resolve => promiseResolve = resolve);

  const promise1 = quxLoader.load();
  promise1.then(([x]) => expect(x.test).toBeTruthy());
  const promise2 = quxLoader.load();
  promise2.then(([x])  => expect(x.test).toBeTruthy());
  promiseResolve({test: true});

  return Promise.all([promise1, promise2]);
});
```

This causes the following eslint errors:

```
  68:3  error  Promise should be returned to test its fulfillment or rejection  jest/valid-expect-in-promise
  70:3  error  Promise should be returned to test its fulfillment or rejection  jest/valid-expect-in-promise
```

It'd be great if the rule would recognize `Promise.all` and considered it valid.

**Comment by SimenB** [comment]
You don't wait for them though - when you do `.then` you return a new promise (from `promise1` and `promise2`) which you _don't_ wait for.

```js
const somePromise = Promise.resolve();

somePromise.then(() => Promise.reject('oh no'));

Promise.all([somePromise]).then(() => {
  console.log('this resolves');
});

// logs
// this resolves
// "Uncaught (in promise) oh no"
```

**Comment by realityking** [comment]
🤦‍♂️ Good catch!

But if I change things to consider the new promises, it still fails:

New code:

```
test('Resource loader: client usage', function () {
  expect.assertions(2);

  const quxLoader = ResourceLoader.create(['qux']);
  let promiseResolve;
  fetchResult = new Promise(resolve => promiseResolve = resolve);

  const promise1 = quxLoader.load()
    .then(([x]) => expect(x.test).toBeTruthy());
  const promise2 = quxLoader.load()
    .then(([x]) => expect(x.test).toBeTruthy());
  promiseResolve({test: true});

  return Promise.all([promise1, promise2]);
});
```

eslint report:

```
  67:9  error  Promise should be returned to test its fulfillment or rejection  jest/valid-expect-in-promise
  69:9  error  Promise should be returned to test its fulfillment or rejection  jest/valid-expect-in-promise
```

**Comment by SimenB** [comment]
Yeah, that's a bug 🙂 Doesn't seem like the rule handles `Promise.all` at all. PR welcome! 😀
